numba/np/arrayobj.py

```diff
@@ -1993,6 +1993,53 @@ def array_repeat_impl(a, repeats):
     return array_repeat_impl
 
 
+@overload(np.tile)
+def np_tile(a, reps):
+    def np_tile_integer_impl(a, reps):
+        if reps < 0:
+            raise ValueError("tile(): negative dimensions are not allowed")
+        arr = np.asarray(a)
+        sz = arr.size
+        out = np.empty(reps * sz, dtype=arr.dtype)
+        arr_flat = arr.flatten()
+        for i in range(reps):
+            start = i * sz
+            end = start + sz
+            out[start:end] = arr_flat
+        return out
+
+    def np_tile_impl(a, reps):
+        for r in reps:
+            if r < 0:
+                raise ValueError("tile(): second argument must be "
+                                 "non-negative")
+
+        arr = np.asarray(a)
+        sz = arr.size
+        # assumes arr is 1-dimensional
+        out_shape = reps[:-1] + (reps[-1] * sz,)
+        num_copies = reduce(lambda x, y: x * y, reps, 1)
+        out = np.empty(shape=num_copies * sz, dtype=arr.dtype)
+        for i in range(num_copies):
+            start = i * sz
+            end = start + sz
+            out[start:end] = arr.flatten()
```

Comment:
call `arr.flatten()` once before the loop

Reply:
Oops, meant to change both implementations but only changed one.